### PR TITLE
Fix spotlessFiles parameter tests

### PR DESCRIPTION
**Pull request by driv**
- `(`,`|` and `)` should not be escaped with `\`, otherwise they get
  interpreted as literals and no file matches
- On Windows paths use backslashes, so `\\\\` has to be used.
2 for java strings and 2 since this parameter is treated as a regex.

Few remarks:
- No issue for this, I just couldn't make it run `check` on Windows. Changes only on Tests.
- I don't have a unix machine right now to test it on.
- I'm not sure if/how it was even running for unix. `.*/src/main/java/test\\(1\\|3\\).java` should have been interpreted as literal.

### CODE REVIEW

**Review by nedtwigg — CHANGES_REQUESTED** [review]
Hmmm... the unix tests were passing, but now they are failing.  I see your point about the parens being interpreted as literal, I'm not sure why this worked before, but it did.

**Comment by nedtwigg** [comment]
Also, needs a changelog entry in `plugin-maven/CHANGES.md`. Can use this PR as the link.

**Comment by driv** [comment]
Any idea what's going on with the npm one? Does it use a different java version?

**Comment by nedtwigg** [comment]
Spotless is JRE 8+
https://github.com/diffplug/spotless/blob/23f8d694a0c93b9d2c5a2343f34b4ffe9ccaeab5/.circleci/config.yml#L92-L107

The Path method you are using is present in [JRE 11](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Path.html#of(java.net.URI)), but not in [JRE 8](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html).

**Comment by nedtwigg** [comment]
Thanks for the fix!

File: plugin-maven/src/test/java/com/diffplug/spotless/maven/SpecificFilesTest.java [hunk 1]
```diff
@@ -16,17 +16,29 @@
...
+		String result = path.toString();
+		if (!isOnWindows()) {
+			return result;
+		} else {
+			return result.replace("\\", "\\\\");
+		}
+	}
+
+	private boolean isOnWindows() {
+		return System.getProperty("os.name").startsWith("Windows");
```
[the remark this answers is not in the document]
Comment (driv) [reply plugin-maven/src/test/java/com/diffplug/spotless/maven/SpecificFilesTest.java:42]: I was 2 seconds away from pushing this change. :)

[the remark this answers is not in the document]
Comment (nedtwigg) [reply plugin-maven/src/test/java/com/diffplug/spotless/maven/SpecificFilesTest.java:42]: Sorry, guess my review was sloppy. "Is the project better now?" If the answer is yes, then I try to get people's stuff merged-in ASAP - I try to err on the side of not blocking people once I'm confident that it's not making new bugs or vulnerabilities. It's not a big deal to have this duplicated. If you want to swap this out in some future PR then fine, if it stays duplicated that's fine too.

[the remark this answers is not in the document]
Comment (driv) [reply plugin-maven/src/test/java/com/diffplug/spotless/maven/SpecificFilesTest.java:42]: Sure! I'll look into the this issue https://github.com/diffplug/spotless/issues/710 and add it there.

**Comment by driv** [comment]
> 
> 
> Thanks for the fix!

Thanks for your support! You make it really easy for people to contribute.

**Pull request merged**